rocksdb_admin/admin_handler.cpp

```diff
@@ -1070,7 +1083,17 @@ void AdminHandler::async_tm_startMessageIngestion(
 
     // TODO: add the logic to write the key/value to rocksdb here
 
-    // TODO: periodically write the message timestamp to local_meta_db
+    // Update meta_db with kafka message timestamp periodically.
+    if (!is_replay && (message_count % FLAGS_kafka_ts_update_interval == 0)) {
```

Comment:
There can be a case that while we are doing replay and the DB was down unexpectedly, in which case we will need to start over replaying when DB is up. A DB write of once every thousand to set a milestone sounds a good balance to me, even while we are in replay mode (any way we are writing DB all the time, this is just 0.1% overhead)

Reply:
Sounds good, I feel I was over optimizing for the catch up phase performance. Will remove it. Thanks for the feedback.